### PR TITLE
Feature/drop support for php 7.3

### PR DESCRIPTION
**Pull request by Chris53897**
https://github.com/doctrine-extensions/DoctrineExtensions/issues/2635

- drop support for php < 7.3
- add property types for final classes
- Remove 2 function related to PHPunit < 9.1
- use null coalescing operator 2x

Needs a check for BC breaks.

### CODE REVIEW

**Review by stof — APPROVED** [review]

**Review by phansys — CHANGES_REQUESTED** [review]
This change should be supported by Rector:
https://github.com/doctrine-extensions/DoctrineExtensions/blob/070f350987f76f12a0f63b82fdfe92b48123e279/rector.php#L24

**Comment by Chris53897** [comment]
Using rector as you suggested will introduce a lot of BC Breaks.

@TomasVotruba Is there a safe way to autoupdate with rector for libraries without BC Breaks?

I would suggest to wait until the problems with doctrine/orm 2.16 are fixed.
After that, merge this PR as it is, and improve after that in follow up PRs.

**Review by phansys — APPROVED** [review]
Maybe we could also implement the rule set [`@PHP74Migration`](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/master/doc/ruleSets/PHP74Migration.rst#rule-set-php74migration) from PHP-CS-Fixer.

**Comment by franmomu** [comment]
Thanks @Chris53897!

**Pull request merged**
